src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -91,10 +91,9 @@ FLAC__StreamDecoderLengthStatus streamLength(const FLAC__StreamDecoder*, FLAC__u
 {
     auto* data = static_cast<sf::priv::SoundFileReaderFlac::ClientData*>(clientData);
 
-    const std::int64_t count = data->stream->getSize();
-    if (count >= 0)
+    if (const auto count = data->stream->getSize())
```

Comment:
```suggestion
    if (const auto count = data->stream->getSize(); count.has_value())
```

Reply:
@ChrisThrasher: The reason why I suggested this change (and similar ones) is that if I am just reading this line of code in isolation I will assume that `getSize()` returns a number and that `count` is just a number. Nothing in the name of the variable nor the name of the function suggests optionality.

I would rather read 

```cpp
if (const auto count = data->stream->getSize(); count.has_value())
```

or 

```
if (const std::optional count = data->stream->getSize())
```

or 

```cpp
if (const std::optional<std::size_t> count = data->stream->getSize())
```

Reply:
Ditto